### PR TITLE
Image was added to the Ontodia element

### PR DESCRIPTION
**Pull request by yuricus**

### CODE REVIEW

File: src/ontodia/data/sparql/provider.ts [hunk 1]
```diff
@@ -17,6 +18,8 @@ const DEFAULT_PREFIX =
  PREFIX rdf:  <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
  PREFIX owl:  <http://www.w3.org/2002/07/owl#>` + '\n\n';
 
+const HAS_MAIN_REPRESENTATION_URI = 'http://collection.britishmuseum.org/id/ontology/PX_has_main_representation';
```
Comment (yuricus) [inline src/ontodia/data/sparql/provider.ts]: We don't need to have any ontology-related resource in source, code, please, move it do initialization of sparql provider in demo.

File: src/ontodia/data/sparql/sparqlModels.ts [hunk 2]
```diff
@@ -44,6 +44,12 @@ export interface ConstructElement {
     object: Field;
 }
 
+export interface ImageLink {
+    inst: Field;
+    type: Field;
+    inst2: Field;
```
Comment (yuricus) [inline src/ontodia/data/sparql/sparqlModels.ts]: Please be more specific. What is inst2? If it's image URL, you should call property as such.

File: src/ontodia/data/sparql/provider.ts (hunk 1 above)
Comment (yuricus) [inline src/ontodia/data/sparql/provider.ts]: Test comment.

File: src/svgui/svgui.ts [hunk 3]
```diff
@@ -86,7 +88,7 @@ export interface Margin {
  *     height: number
  *     margin: svgui.Margin
  */
-export abstract class UIElement extends Backbone.Model {
```
Comment (AlexeyMz) [inline src/svgui/svgui.ts]: Why remove `abstract` from `UIElement`?

File: src/svgui/svgui.ts [hunk 4]
```diff
@@ -96,7 +98,7 @@ export abstract class UIElement extends Backbone.Model {
     }
     defaults(): any {
         return {
-            margin: {top: 0, right: 0, bottom: 0, left: 0},
+            margin: { top: 0, right: 0, bottom: 0, left: 0 },
```
Comment (AlexeyMz) [inline src/svgui/svgui.ts]: The spaces are unnecessary here. See `STYLEGUIDE.md` for details.

File: src/svgui/svgui.ts [hunk 5]
```diff
@@ -3,6 +3,8 @@ import * as Backbone from 'backbone';
 import * as _ from 'lodash';
 import * as $ from 'jquery';
 
+const IMAGE_MIN_SIZE: Vector = vector(100, 100);
```
Comment (AlexeyMz) [inline src/svgui/svgui.ts]: Minimum image size should not be a part of `svgui` but provided as parameter from outside.

File: src/svgui/svgui.ts [hunk 6]
```diff
@@ -250,6 +251,69 @@ export class Pair extends UIElement {
     }
 }
 
+/*
+ * Image for element.
+ * Backbone properties:
+ *     imageUrl: string
+ */
+export class Image extends UIElement {
```
Comment (AlexeyMz) [inline src/svgui/svgui.ts]: Remove all other unnecessary changes in this file (except for adding `Image` class), please.

File: src/svgui/svgui.ts [hunk 7]
```diff
@@ -250,6 +251,69 @@ export class Pair extends UIElement {
...
+    public image: d3.Selection<any>;
+    public imageRect: d3.Selection<any>;
+
+    defaults() {
+        return _.extend(super.defaults(), {
+            imageUrl: undefined,
+            borderColor: 'green',
+            spacing: vector(10, 3),
```
Comment (AlexeyMz) [inline src/svgui/svgui.ts]: `spacing` attribute is unused.

File: src/svgui/svgui.ts [hunk 8]
```diff
@@ -250,6 +251,69 @@ export class Pair extends UIElement {
...
+            padding: undefined,
```
Comment (AlexeyMz) [inline src/svgui/svgui.ts]: Do we really need separate `padding` attribute in addition to `margin`?

File: src/svgui/svgui.ts [hunk 9]
```diff
@@ -250,6 +251,69 @@ export class Pair extends UIElement {
...
+            .attr('stroke', this.get('borderColor'));
+        this.image = this.root.append('image')
+            .attr('x', 0)
+            .attr('y', 0)
+            .attr('width', this.get('minSize').x)
+            .attr('height', this.get('minSize').y)
+            .attr('href', this.get('imageUrl'));
+    }
+    update() {
+        if (this.image) {
```
Comment (AlexeyMz) [inline src/svgui/svgui.ts]: You could remove these `if` statements by moving `this.update()` call after setting `this.imageRect` and `this.image` fields in `initialize()`.

File: src/svgui/svgui.ts [hunk 10]
```diff
@@ -250,6 +251,69 @@ export class Pair extends UIElement {
...
+            padding: undefined,
+            minSize: vector(100, 100),
+        });
+    }
+    initialize() {
+        super.initialize.apply(this, arguments);
+        this.root.attr('class', 'svguiImage');
+        this.update();
+        this.imageRect = this.root.append('rect')
+            .attr('fill', '#e1e1e1')
```
Comment (AlexeyMz) [inline src/svgui/svgui.ts]: Consider adding `background` attribute with default value `#e1e1e1`.

File: src/svgui/svgui.ts [hunk 11]
```diff
@@ -250,6 +251,69 @@ export class Pair extends UIElement {
     }
 }
 
+/*
+ * Image for element.
+ * Backbone properties:
+ *     imageUrl: string
```
Comment (AlexeyMz) [inline src/svgui/svgui.ts]: Some backbone properties (like `minSize`, ...) are missing from documentation.

File: src/svgui/svgui.ts (hunk 8 above)
Comment (Ladone3) [inline src/svgui/svgui.ts]: We can set as default padding (1px,1px,1px,1px) otherwise the image will be overlap the border line.
